### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by CarlosE-Feliz**
- Create a class Corrector. ✔
- Set an instance of Corrector in Person. ✔
- In Person add a method validate_name. ✔

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
+require './corrector'
 # Create class person
-class Person
+class Person < Corrector
```
Comment (eliasecasta) [inline person.rb]: - [x] Person should not be based on the corrector class, there should be an instance of the corrector class set in the person class on its initialization like this:

```ruby
class Person
  .
  .

  def initialize(age, name = "Unknown", parent_permission = true)
    .
    @corrector = Corrector.new()
    .
    .
  end

  .
  .
end
```

Please make this small change to comply with the project requirements.

[Support article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/composition_by_examples.md#composable-classes)

File: person.rb [hunk 2]
```diff
@@ -19,4 +21,8 @@ def age_of?
   def can_use_service?
     age_of? || @parent_permission
   end
+
+  def validate_name
+    /[[:upper:]]/.match(@name)
+  end
```
Comment (eliasecasta) [inline person.rb]: - [x] The validate name method must use the corrector instance to validate and save the corrected name, please do so to comply with the project requirements.

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @xJokeer, I am Elias, your reviewer for this PR

Well done so far with your project!

Your project seems good so far, almost everything is correctly set up and you are only missing some really small changes, keep it up!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [x] Please improve your Readme, make it more descriptive to comply with the [general requirements.](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- [x] Please check the comments below for more requirements.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb (hunk 1 above)
Comment (CarlosE-Feliz) [reply person.rb]: Thanks for the feedback! to be honest i was a bit confused about that part!

**Review by EricMbouwe — APPROVED** [review]
### Hi @xJokeer, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**